index.js

```diff
@@ -1,11 +1,17 @@
 'use strict';
 
 var traverse = module.exports = function (schema, opts, cb) {
+  // Legacy support for v0.3.1 and earlier.
   if (typeof opts == 'function') {
     cb = opts;
     opts = {};
   }
-  _traverse(opts, cb, schema, '', schema);
+
+  cb = opts.cb || cb;
+  var pre = (typeof cb == 'function') ? cb : cb.pre || function() {};
```

Comment:
cb must be passed one way or another, so supplying an empty function here would hide an issue... I will remove it. Otherwise looks good.

Reply:
My thinking was you could call `traverse(schema, {cb: {post: () => ...}})` if you only want to do something in the post-traverse and don't care about the pre-traverse.

Reply:
I see... All right then.